projects/Ethersync/programs/ethersync/tests/basic.nix

```diff
@@ -0,0 +1,79 @@
+{
+  sources,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  name = "Ethersync";
+
+  interactive.sshBackdoor.enable = true;
+
+  nodes =
+    let
+      config = {
+        imports = [
+          sources.modules.ngipkgs
+          sources.modules.programs.ethersync
+          sources.examples.Ethersync.basic
+        ];
+      };
+    in
+    {
+      server = config // {
+        networking.firewall.allowedTCPPorts = [ 4242 ];
+        networking.interfaces.eth1.ipv4.addresses = lib.mkForce [
+          {
+            address = "192.168.1.1";
+            prefixLength = 24;
+          }
+        ];
+      };
+      client = config // {
+        services.getty.autologinUser = "root";
+        networking.interfaces.eth1.ipv4.addresses = lib.mkForce [
+          {
+            address = "192.168.1.2";
+            prefixLength = 24;
+          }
+        ];
+      };
+    };
+
+  testScript =
+    let
+      key = "CAESQGurlr9XTdGuz2nXI6esucINWpDoLBIW2qlYhOKGrggLrd9aLlCdAp1iQE6ZEMzFFVv5KQXp7sTB+YhllBZ/NgQ=";
+      peer = "12D3KooWMX6EYnLXfWs3vsGhtWgqBfRSoWqs4GiDMu3rB4qSjRpX";
+      script =
+        let
+          esc = lib.strings.fromJSON ''"\u001B"'';
+        in
+        pkgs.writeText "script" "ddiclient${esc}";
+    in
+    ''
+      import time
+
+      start_all()
+
+      server.wait_for_unit("default.target")
+      server.succeed("mkdir -p .ethersync")
+      server.succeed("echo ${key} | base64 -d >.ethersync/key")
+      server.succeed("chmod 600 .ethersync/key")
+      server.succeed("echo server >file.txt")
+      server.execute("ethersync daemon --port 4242 >/dev/null &")
+      server.wait_for_open_port(4242)
+
+      client.wait_for_unit("default.target")
+      client.succeed("mkdir -p .ethersync")
```

Comment:
Is that used anywhere?

Reply:
Can we have this as a comment? like: `# enable ethersync for current directory`